### PR TITLE
AV-68687: Code for Nextjs Quickstart

### PR DESCRIPTION
**Pull request by teetangh**

### CODE REVIEW

**Review by prajwal-pai77 — CHANGES_REQUESTED** [review]
Do we need a UI which shows the nextJS swagger documentation and github link?

I get an error in the airport route, can you please check?
image
Additionally can you standardize the responses, request body which is on the swagger UI by looking at the other tutorials?

File: app/api/v1/airline/[airlineId]/route.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
+import { NextRequest, NextResponse } from "next/server"
+
+import { getDatabase } from "@/lib/couchbase-connection"
+import { TAirline,AirlineSchema } from "@/app/models/Airline"
+import { DocumentNotFoundError,DocumentExistsError } from 'couchbase';
+
+import { ZodError } from "zod";
+
+/**
+ * @swagger
```
Comment (prajwal-pai77) [inline app/api/v1/airline/[airlineId]/route.ts:10]: Why do i get No links on the right hand side of the swagger response on the UI?
Can you pls remove it?
image

File: app/api/v1/airline/[airlineId]/route.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+ *   get:
+ *     summary: Get an airline by ID
+ *     description: |
+ *       Get Airline with specified ID.
+ *       
+ *       This provides an example of using [Key Value operations](https://docs.couchbase.com/nodejs-sdk/current/howtos/kv-operations.html) in Couchbase to get a document with specified ID.
+ *       
+ *       Key Value operations are unique to Couchbase and provide very high-speed get/set/delete operations.
+ *       
+ *       Code: `airline/[airlineId]/route.ts` Method: `GET`
```
Comment (prajwal-pai77) [inline app/api/v1/airline/[airlineId]/route.ts:21]: Please add a break line between Code and method.
Same for all swagger route.

File: app/api/v1/airline/[airlineId]/route.ts [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+ *     tags:
+ *        - Airline
+ *     parameters:
+ *       - name: airlineId
+ *         in: path
+ *         description: ID of the airline
```
Comment (prajwal-pai77) [inline app/api/v1/airline/[airlineId]/route.ts:27]: Can you please update this to: Airline ID like airline_10

File: app/api/v1/airline/[airlineId]/route.ts [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+ *       
+ *       Key Value operations are unique to Couchbase and provide very high-speed get/set/delete operations.
+ *       
+ *       Code: `airline/[airlineId]/route.ts` Method: `POST`
+ *     tags:
+ *        - Airline
+ *     parameters:
+ *       - name: airlineId
+ *         in: path
+ *         description: ID of the airline
```
Comment (prajwal-pai77) [inline app/api/v1/airline/[airlineId]/route.ts:84]: Same as above.
Similarly please do it for other routes as well.

File: app/api/v1/airline/to-airport/route.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+ *    
+ *       This provides an example of using SQL++ query in Couchbase to fetch a list of documents matching the specified criteria.
+ * 
+ *       Code: [`app/api/v1/airline/to-airport/route.ts`] Method: `GET`
+ *     tags:
+ *        - Airline
+ *     parameters:
+ *       - name: destinationAirportCode
+ *         in: query
+ *         description: The ICAO or IATA code of the destination airport
```
Comment (prajwal-pai77) [inline app/api/v1/airline/to-airport/route.ts:23]: Can you standardize this by looking at other tutorials swagger UI, basically it also includes examples as well.
Similarly for all routes.

**Review by prajwal-pai77 — CHANGES_REQUESTED** [review]
Do we need a UI which shows the nextJS swagger documentation and github link?
I get an error in the airport route, can you please check?
image
Additionally can you standardize the responses, request body which is on the swagger UI by looking at the other tutorials?

File: public/vercel.svg
[the remark this answers is not in the document]
Comment (teetangh) [reply public/vercel.svg:1]: done

File: app/api/hello/route.ts
[the remark this answers is not in the document]
Comment (teetangh) [reply app/api/hello/route.ts:1]: removed

File: lib/couchbase-connection.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
+import * as couchbase from "couchbase"
+
+const DB_USERNAME: string = process.env.DB_USERNAME ?? "kaustav"
```
[the remark this answers is not in the document]
Comment (teetangh) [reply lib/couchbase-connection.ts:3]: changed

File: lib/couchbase-connection.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+if (!DB_USERNAME) {
+  throw new Error(
+    "Please define the DB_USERNAME environment variable inside .env.local"
+  )
+}
+
+if (!DB_PASSWORD) {
+  throw new Error(
+    "Please define the DB_PASSWORD environment variable inside .env.local"
```
[the remark this answers is not in the document]
Comment (teetangh) [reply lib/couchbase-connection.ts:16]: added the example

File: app/api/v1/airline/[airlineId]/route.ts (hunk 3 above)
Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:27]: updated in all

Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:27]: image

Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:27]: added examples in all possible routes

File: app/api/v1/airline/[airlineId]/route.ts (hunk 4 above)
Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:84]: added examples in all routes

File: app/api/v1/airline/[airlineId]/route.ts (hunk 1 above)
Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:10]: There is no currently no way to remove the "no links" from the nextjs swagger package.We can only add links

File: app/api/v1/airline/to-airport/route.ts (hunk 5 above)
Comment (teetangh) [reply app/api/v1/airline/to-airport/route.ts:23]: have included the examples

File: app/api/v1/airline/[airlineId]/route.ts [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+    const { airlineCollection } = await getDatabase()
+
+    const updatedAirline = await airlineCollection.upsert(
+      airlineId,
+      parsedAirlineData
+    )
+    return NextResponse.json(
+      {
+        airlineId: airlineId,
+        airlineData: parsedAirlineData,
```
[the remark this answers is not in the document]
Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:209]: Have standardized it

File: app/api/v1/airline/[airlineId]/route.ts (hunk 2 above)
Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.ts:21]: added everywhere

**Review by prajwal-pai77 — CHANGES_REQUESTED** [review]
Couple of changes majorly on fetching credentials from environment file.

File: app/api/v1/airline/[airlineId]/route.test.ts [hunk 9]
```diff
@@ -0,0 +1,230 @@
+import { NextRequest } from "next/server"
```
[the remark this answers is not in the document]
Comment (teetangh) [reply app/api/v1/airline/[airlineId]/route.test.ts:1]: These are not errors. These are warnings and are present in spring boot as well. Just making sure that these dummy data are not mixed with real data before and after the test

File: .env.local.example [hunk 10]
```diff
@@ -0,0 +1,4 @@
+DB_USERNAME=Administrator
```
[the remark this answers is not in the document]
Comment (teetangh) [reply .env.local.example:1]: Its fetching from .env and not .env.local

File: lib/couchbase-connection.ts [hunk 11]
```diff
@@ -0,0 +1,99 @@
+import * as couchbase from "couchbase"
+
+const DB_USERNAME: string = process.env.DB_USERNAME ?? "Administrator"
```
[the remark this answers is not in the document]
Comment (teetangh) [reply lib/couchbase-connection.ts:3]: nope. in the spring boot, kotlin quickstarts,etc. considering the user forgets to put the creds in the dev.env file or doesnt even create the file. We can atlteast help him with the default scenario.

File: README.md [hunk 12]
```diff
@@ -1 +1,161 @@
...
+
+![Application Startup](./public/app_startup.png)
+
+The application will run on the port specified by Nextjs on your local machine (eg: http://localhost:3000). You will find the interactive Swagger documentation of the API if you go to the [URL](http://localhost:3000/api-doc) in your browser. Swagger documentation is used in this demo to showcase the different API endpoints and how they can be invoked. More details on the Swagger documentation can be found in the [appendix](#swagger-documentation).
+
+![Swagger Documentation](./public/swagger_documentation.png)
+
+## Running The Tests
+
+To run the integration tests, use the following commands:
```
[the remark this answers is not in the document]
Comment (teetangh) [reply README.md:109]: okay. mentioned

File: app/models/Airline.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import { z } from 'zod';
+
+const AirlineSchema = z.object({
+  callsign: z.string().optional(),
+  country: z.string(),
+  iata: z.string().optional(),
+  icao: z.string(),
+  id: z.number(),
```
[the remark this answers is not in the document]
Comment (teetangh) [reply app/models/Airline.ts:8]: changed

File: app/models/Airline.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import { z } from 'zod';
+
+const AirlineSchema = z.object({
+  id: z.number().optional(),
```
Comment (prajwal-pai77) [inline app/models/Airline.ts:4]: Why am i still seeing the id and type field in both swagger and the output?
image
Also can you prefill the fields with some default values like we do for other tutorials?

**Review by prajwal-pai77 — CHANGES_REQUESTED** [review]

File: app/models/Airline.ts (hunk 14 above)
Comment (teetangh) [reply app/models/Airline.ts:4]: id and type are optional only.
image
The user can choose not input those fields

Comment (prajwal-pai77) [reply app/models/Airline.ts:4]: Please remove it as they are confusing.
See [requirements](https://docs.google.com/document/d/1_uc_HRKaNSAcEUsUgdAR9EamIotcO4LwOAJZAcMi7fk/edit?pli=1#heading=h.qpuhsn242jgu)

Comment (teetangh) [reply app/models/Airline.ts:4]: image
Filled in with examples

**Review by prajwal-pai77 — APPROVED** [review]

**Pull request merged**